lib/jekyll/commands/doctor.rb

```diff
@@ -44,6 +45,29 @@ def healthy?(site)
           ].all?
         end
 
+        def outdated_dependency_check(site)
+          return true unless can_test_dependency_health?(site)
+
+          Dir.chdir(site.source) do
+            exe = Gem.bin_path("bundler", "bundle")
+            process, output = Jekyll::Utils::Exec.run(
+              "ruby", exe, "outdated", "--parseable"
+            )
+
+            raise SystemExit unless process.success?
+            return true if output.empty?
+
+            output.to_s.each_line do |line|
+              Jekyll.logger.warn "Outdated gem:", line
+            end
+
+            Jekyll.logger.info "", "Found outdated site dependencies!".magenta
+            Jekyll.logger.info "", "Run `bundle update` to use latest " \
+                                   "versions of those gems in your site".magenta
```

Comment:
I like the idea of encouraging users to upgrade.

Now if I run `bundle update` I don't necessarily the latest versions installed, depends also on the version specified in the `Gemfile`, and `.gemspec`, right?

Here's Jekyll current output if I check for outdated gems:

```sh
bundle outdated --parseable

activesupport (newest 5.1.4, installed 4.2.10)
coffee-script-source (newest 1.12.2, installed 1.11.1)
rainbow (newest 3.0.0, installed 2.2.2)
rdoc (newest 6.0.1, installed 5.1.0, requested ~> 5.0)
rubocop (newest 0.52.1, installed 0.51.0, requested ~> 0.51.0)
shoulda-matchers (newest 3.1.2, installed 2.8.0)
simplecov (newest 0.15.1, installed 0.13.0)
```

So here even after running `bundler update`, `jekyll doctor`would still encourage to run `bundle update`, it could be confusing don't you think? 😕

Reply:
ohh.. I did not expect Bundler to output when the `Gemfile` *locks* to a certain version series..
hmm.. thanks for pointing this out.. :+1:

Reply:
`jekyll doctor` could compare the result of running `bundle update && bundle outdated`--parseable` vs `bundle outdated --parseable` and only warn users if the output differs?

Reply:
But don't run `bundle update` because that might result unintentionally upgrading